nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/ListenTrapSNMP.java

```diff
@@ -0,0 +1,119 @@
+import org.apache.nifi.snmp.operations.SNMPTrapReceiver;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+public class ListenTrapSNMP extends AbstractSNMPProcessor {
+
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("All FlowFiles that are received from the SNMP agent are routed to this relationship")
+            .build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("All FlowFiles that cannot received from the SNMP agent are routed to this relationship")
+            .build();
+
+    protected static final List<PropertyDescriptor> PROPERTY_DESCRIPTORS = Collections.unmodifiableList(Arrays.asList(
+            SNMP_CLIENT_PORT,
+            AGENT_HOST,
+            AGENT_PORT,
+            SNMP_VERSION,
+            SNMP_COMMUNITY,
+            SNMP_SECURITY_LEVEL,
+            SNMP_SECURITY_NAME,
+            SNMP_AUTH_PROTOCOL,
+            SNMP_AUTH_PASSWORD,
+            SNMP_PRIVACY_PROTOCOL,
+            SNMP_PRIVACY_PASSWORD,
+            SNMP_RETRIES,
+            SNMP_TIMEOUT
+    ));
+
+    private static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+            REL_SUCCESS,
+            REL_FAILURE
+    )));
+
+    private SNMPTrapReceiver trapReceiver;
+
+    @OnScheduled
+    @Override
+    public void initSnmpClient(ProcessContext context) {
+        final String clientPort = context.getProperty(SNMP_CLIENT_PORT).getValue();
+
+        final TargetConfiguration configuration = new TargetConfigurationBuilder()
+                .setVersion(context.getProperty(SNMP_VERSION).getValue())
+                .setAuthProtocol(context.getProperty(SNMP_AUTH_PROTOCOL).getValue())
+                .setAuthPassword(context.getProperty(SNMP_AUTH_PASSWORD).getValue())
+                .setPrivacyProtocol(context.getProperty(SNMP_PRIVACY_PROTOCOL).getValue())
+                .setPrivacyPassword(context.getProperty(SNMP_PRIVACY_PASSWORD).getValue())
+                .setSecurityName(context.getProperty(SNMP_SECURITY_NAME).getValue())
+                .setSecurityLevel(context.getProperty(SNMP_SECURITY_LEVEL).getValue())
+                .createSecurityConfiguration();
+
+        snmpClient = SNMPClientFactory.createSnmpClient(configuration, clientPort);
+        try {
+            snmpClient.listen();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @OnStopped
+    public void stopSnmpClient(ProcessContext context) {
+        trapReceiver = null;
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) {
+        if (Objects.isNull(trapReceiver)) {
```

Comment:
Is there a chance that this is null after onScheduled?

Reply:
Yes, there's no @OnScheduled call. I must have initialized this here to pass the context and session.